File: frontend/src/types/system-info.ts
export interface ModelInfo {
  name: string;
  family: string | null;
  parameter_size: string | null;
  quantization: string | null;
  context_length: number | null;
  experts: string | null;
  format: string | null;
}

export interface OllamaInfo {
  version: string | null;
  model_loaded: boolean;
  model_size_gb: number | null;
  vram_used_gb: number | null;
  gpu_layers: number | null;
  gpu_count: number | null;
}

export interface WorkerInfo {
  celery_active: boolean;
  task_timeout: number;
  concurrency: string | null;
}

export interface GpuDevice {
  name: string;
  memory_total_mb: number;
  memory_used_mb: number | null;
}

export interface GpuInfo {
  driver_version: string | null;
  devices: GpuDevice[];
}

export interface SystemInfo {
  model: ModelInfo;
  source: string;
  ollama: OllamaInfo | null;
  gpu: GpuInfo | null;
  worker: WorkerInfo;
  timeout_seconds: number;
  version: string;
}

File: backend/app/schemas/system_info.py
from pydantic import BaseModel


class ModelInfo(BaseModel):
    name: str
    family: str | None = None
    parameter_size: str | None = None
    quantization: str | None = None
    context_length: int | None = None
    experts: str | None = None  # e.g. "8/256 active"
    format: str | None = None


class OllamaInfo(BaseModel):
    version: str | None = None
    model_loaded: bool = False
    model_size_gb: float | None = None
    vram_used_gb: float | None = None
    gpu_layers: int | None = None
    gpu_count: int | None = None


class WorkerInfo(BaseModel):
    celery_active: bool = False
    task_timeout: int = 600
    concurrency: str | None = None


class GpuDevice(BaseModel):
    name: str
    memory_total_mb: int
    memory_used_mb: int | None = None


class GpuInfo(BaseModel):
    driver_version: str | None = None
    devices: list[GpuDevice] = []


class SystemInfoResponse(BaseModel):
    model: ModelInfo
    source: str  # "Ollama", "Anthropic", "OpenAI", etc.
    ollama: OllamaInfo | None = None
    gpu: GpuInfo | None = None
    worker: WorkerInfo
    timeout_seconds: int = 600
    version: str = "0.1.0"
